qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -133,86 +141,128 @@ def circuits(self) -> List[QuantumCircuit]:
-                circuits.append(rb_circ)
+
+            # Compute inverse, compute only the difference from the previous shorter sequence
+            for elem in seq[len(prev_seq) :]:
+                prev_elem = self.__compose_clifford(prev_elem, elem)
+            prev_seq = seq
+            inv = self.__adjoint_clifford(prev_elem)
+
+            circ.append(self._to_instruction(inv), qubits)
+            circ.barrier(qubits)  # TODO: Can we remove this? (measure_all inserts one more barrier)
+            circ.measure_all()
+            circuits.append(circ)
         return circuits
 
+    def _sample_sequence(self, length: int, rng: Generator) -> Sequence[SequenceElementType]:
+        """Sample a RB sequence with the given length.
+
+        Args:
+            length: RB sequences length.
+            rng: Generator object for random number generation.
+
+        Returns:
+            A RB sequence.
+        """
+        # Return integer instead of Clifford object for 1 or 2 qubit case
+        if self.num_qubits == 1:
+            return rng.integers(24, size=length)
+        if self.num_qubits == 2:
+            return rng.integers(11520, size=length)
+
+        return [random_clifford(self.num_qubits, rng) for _ in range(length)]
+
+    def _to_instruction(self, elem: SequenceElementType) -> Instruction:
+        # TODO: basis transformation in 1Q (and 2Q) cases for speed
+        # Switching for speed up
+        if isinstance(elem, Integral):
+            if self.num_qubits == 1:
+                return _clifford_1q_int_to_instruction(elem)
+            if self.num_qubits == 2:
+                return _clifford_2q_int_to_instruction(elem)
+        return elem.to_instruction()
+
+    def __identity_clifford(self) -> SequenceElementType:
+        if self.num_qubits <= 2:
+            return 0
+        return Clifford(np.eye(2 * self.num_qubits))
+
+    def __compose_clifford(
+        self, lop: SequenceElementType, rop: SequenceElementType
+    ) -> SequenceElementType:
+        # TODO: Speed up 1Q (and 2Q) cases using lookup table
+        if self.num_qubits == 1:
+            if isinstance(lop, Integral):
+                lop = CliffordUtils.clifford_1_qubit(lop)
+            if isinstance(rop, Integral):
+                rop = CliffordUtils.clifford_1_qubit(rop)
+        if self.num_qubits == 2:
+            if isinstance(lop, Integral):
+                lop = CliffordUtils.clifford_2_qubit(lop)
+            if isinstance(rop, Integral):
+                rop = CliffordUtils.clifford_2_qubit(rop)
+        return lop.compose(rop)
+
+    def __adjoint_clifford(self, op: SequenceElementType) -> SequenceElementType:
+        # TODO: Speed up 1Q and 2Q cases using lookup table
+        return op.adjoint()
+
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
+        # TODO: Custom transpilation (without calling transpile()) for 1Q and 2Q cases
```

Comment:
Is it possible to integrate qubit-aware basis gate count into circuit generation? Actually this counting logic inserts another loop and the transpile slower. This count information is only used by the Standard RB (for EPC to EPG conversion), but if the performance regression is quite small I think having basis gate counts in metadata of other types of RB circuits is kind of nice.

Reply:
I don't think the part is currently a bottleneck. It would not be too late to think of that after we clearly see the part is a bottleneck.